include/openthread/netdiag.h

```diff
@@ -128,6 +129,45 @@ typedef struct otNetworkDiagRoute
     otNetworkDiagRouteData mRouteData[OT_NETWORK_MAX_ROUTER_ID + 1]; ///< Link Quality and Routing Cost data.
 } otNetworkDiagRoute;
 
+/**
+ * Represents a Network Diagnostic Enhanced Route data.
+ */
+typedef struct otNetworkDiagEnhRouteData
+{
+    uint8_t mRouterId;           ///< The Router ID.
+    bool    mIsSelf : 1;         ///< This is the queried device itself. If set, the other fields should be ignored.
+    bool    mHasLink : 1;        ///< Indicates whether the queried device has a direct link with router.
+    uint8_t mLinkQualityOut : 2; ///< Link Quality Out (applicable when `mHasLink`).
+    uint8_t mLinkQualityIn : 2;  ///< Link Quality (applicable when `mHasLink`).
```

Comment:
Should this also say "Ïn"?
```suggestion
    uint8_t mLinkQualityIn : 2;  ///< Link Quality In (applicable when `mHasLink`).
```